src/Controller/EnMarche/CitizenProjectController.php

```diff
@@ -199,4 +199,48 @@ public function listActorsAction(CitizenProject $citizenProject, CitizenProjectM
             'actors' => $citizenProjectManager->getCitizenProjectMemberships($citizenProject),
         ]);
     }
+
+    /**
+     * @Route("/{slug}/rejoindre", name="app_citizen_project_follow", condition="request.request.has('token')")
+     * @Method("POST")
+     * @Security("is_granted('FOLLOW_CITIZEN_PROJECT', citizenProject)")
+     */
+    public function followAction(Request $request, CitizenProject $citizenProject): Response
+    {
+        if (!$this->isCsrfTokenValid('citizen_project.follow', $request->request->get('token'))) {
+            throw $this->createAccessDeniedException('Invalid CSRF protection token to follow citizen project.');
+        }
+
+        $this->get('app.citizen_project.authority')->followCitizenProject($this->getUser(), $citizenProject);
+
+        return new JsonResponse([
+            'button' => [
+                'label' => 'Quitter ce projet citoyen',
```

Comment:
Sounds weird to return view data from the action but perhaps it's done like this on this project.

Reply:
yes, see the same logical on committee follow / unfollow